File: frontend/src/services/listeningLessonReviewService.js
import axiosClient from "../api/axiosClient";

const listeningLessonReviewService = {
  getByLesson(lessonId) {
    return axiosClient.get(`/v1/listening-lesson-reviews/lesson/${lessonId}`);
  },
  create(data) {
    return axiosClient.post("/v1/listening-lesson-reviews", data);
  },

  getAll() {
    return axiosClient.get("/v1/listening-lesson-reviews");
  },
  getById(id) {
    return axiosClient.get(`/v1/listening-lesson-reviews/${id}`);
  },
  delete(id) {
    return axiosClient.delete(`/v1/listening-lesson-reviews/${id}`);
  },
};

export default listeningLessonReviewService;

File: backend/src/main/java/com/englishlearning/backend/service/ListeningLessonReviewService.java
package com.englishlearning.backend.service;

import com.englishlearning.backend.dto.request.ListeningLessonReviewRequest;
import com.englishlearning.backend.dto.response.ListeningLessonReviewResponse;

import java.util.List;

public interface ListeningLessonReviewService {

    ListeningLessonReviewResponse create(ListeningLessonReviewRequest request);

    ListeningLessonReviewResponse getById(Long reviewId);

    List<ListeningLessonReviewResponse> getByLessonId(Long lessonId);

    List<ListeningLessonReviewResponse> getAll();

    void delete(Long reviewId);
}